### PR TITLE
Add hooks for constructing connections to and from modal representations

### PR DESCRIPTION
**Pull request by thomasgibson**
This brings in the recent modal connection work in meshmode to grudge. Can be used to simplify the modal indicator used in https://github.com/illinois-ceesd/mirgecom/pull/213 and modal filtering in https://github.com/illinois-ceesd/mirgecom/pull/92

@MTCam @lukeolson @inducer

### CODE REVIEW

File: grudge/dof_desc.py [hunk 1]
```diff
@@ -58,6 +60,13 @@ class DTAG_VOLUME_ALL:  # noqa: N801
     """
 
 
+class DTAG_MODAL:  # noqa: N801
+    """
+    A tag denoting modal (rather than nodal)
+    degrees of freedom on cell volumes.
+    """
```
Comment (inducer) [inline grudge/dof_desc.py:67]: This is not the right place for this. `DTAG` stands for "domain tag" (as you can tell from the surroundings), i.e. which geometry this is on. `QTAG` ("quadrature tag") describes how something is discretized. It's obviously possible to use modal discretizations on boundaries and volumes, so connecting these two concerns does not make sense.

`QTAG_MODAL` would be better, however "quadrature" is a misnomer. I'm open to renaming `QTAG`.

Comment (thomasgibson) [reply grudge/dof_desc.py:67]: I vote for `QTAG_MODAL` for now.

Comment (thomasgibson) [reply grudge/dof_desc.py:67]: To be fair, `DTAG_SCALAR` doesn't seem to make sense as a "domain tag" either. But I'll leave that for another day.

Comment (inducer) [reply grudge/dof_desc.py:67]: > To be fair, `DTAG_SCALAR` doesn't seem to make sense as a "domain tag" either. But I'll leave that for another day.

Why doesn't it? The domain of the scalar is a set (any set) with a single element.

Comment (thomasgibson) [reply grudge/dof_desc.py]: Fixed in https://github.com/inducer/grudge/pull/77/commits/e9f6da0d85da1802a2b3f9b43ac32c5173ef65d9

Comment (thomasgibson) [reply grudge/dof_desc.py]: > Why doesn't it? The domain of the scalar is a set (any set) with a single element.

I think I was just incorrectly trying to pin too much "fem" type concepts to the tag when I initially looked at this. Disregard what I said. It's my misunderstanding.

Comment (inducer) [reply grudge/dof_desc.py]: #79

File: grudge/discretization.py [hunk 2]
```diff
@@ -239,10 +261,10 @@ def connection_from_dds(self, from_dd, to_dd):
 
         # {{{ generic to-quad
 
-        if (
-                from_dd.domain_tag == to_dd.domain_tag
-                and from_dd.quadrature_tag is QTAG_NONE
-                and to_dd.quadrature_tag is not QTAG_NONE):
+        if (from_dd.domain_tag == to_dd.domain_tag
```
Comment (inducer) [inline grudge/discretization.py:264]:
```suggestion
        # QTAG_MODAL is handled above
        if (from_dd.domain_tag == to_dd.domain_tag
```

File: grudge/dof_desc.py [hunk 3]
```diff
@@ -36,12 +36,14 @@
...
 .. autoclass:: DTAG_BOUNDARY
 .. autoclass:: QTAG_NONE
+.. autoclass:: QTAG_MODAL
 
 .. autoclass:: DOFDesc
 .. autofunction:: as_dofdesc
 
 .. data:: DD_SCALAR
 .. data:: DD_VOLUME
+.. data:: DD_MODAL
```
Comment (inducer) [inline grudge/dof_desc.py:46]:
```suggestion
.. data:: DD_VOLUME_MODAL
```
? I'm also OK with `DD_VOL_MODAL`.

**Review by inducer — COMMENTED** [review]
Thanks! Two minor things, otherwise LGTM.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**